website/docs/r/app_service_virtual_network_connection.html.markdown

```diff
@@ -0,0 +1,169 @@
+Please refer to this doc: https://docs.microsoft.com/en-us/azure/app-service/web-sites-integrate-with-vnet#gateway-required-vnet-integration
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "test" {
+  name     = "test"
+  location = "West Europe"
+}
+
+resource "azurerm_resource_group" "testvnet" {
+  name     = "testvnet"
+  location = "Central US"
+}
+
+resource "azurerm_app_service_plan" "test" {
+  name                = "test"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  sku {
+    tier = "Standard"
+    size = "S1"
+  }
+}
+
+resource "azurerm_app_service" "test" {
+  name                = "test-appservice"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  app_service_plan_id = "${azurerm_app_service_plan.test.id}"
+}
+
+resource "azurerm_virtual_network" "test" {
+  name                = "test-virtual-network"
+  resource_group_name = "${azurerm_resource_group.testvnet.name}"
+  location            = "${azurerm_resource_group.testvnet.location}"
+  address_space       = ["10.0.0.0/16"]
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "GatewaySubnet"
+  resource_group_name  = "${azurerm_resource_group.testvnet.name}"
+  virtual_network_name = "${azurerm_virtual_network.test.name}"
+  address_prefix       = "10.0.1.0/24"
+}
+
+resource "azurerm_public_ip" "test" {
+  name                = "test"
+  location            = "${azurerm_resource_group.testvnet.location}"
+  resource_group_name = "${azurerm_resource_group.testvnet.name}"
+
+  allocation_method = "Dynamic"
+}
+
+resource "azurerm_virtual_network_gateway" "test" {
+  name                = "test"
+  location            = "${azurerm_resource_group.testvnet.location}"
+  resource_group_name = "${azurerm_resource_group.testvnet.name}"
+
+  type     = "Vpn"
+  vpn_type = "RouteBased"
+  sku      = "Basic"
+
+  ip_configuration {
+    name                          = "vnetGatewayConfig"
+    public_ip_address_id          = "${azurerm_public_ip.test.id}"
+    private_ip_address_allocation = "Dynamic"
+    subnet_id                     = "${azurerm_subnet.test.id}"
+  }
+
+  vpn_client_configuration {
+    address_space = ["10.2.0.0/24"]
+  }
+}
+
+
+resource "azurerm_app_service_virtual_network_connection" "test" {
+  resource_group_name   = "${azurerm_resource_group.test.name}"
+  app_service_name      = "${azurerm_app_service.test.name}"
+
+  properties {
+    vnet_resource_id    = "${azurerm_virtual_network.test.id}"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `app_service_name` - (Required) Specifies the name of the App Service.
+
+* `resource_group_name` - (Required) The name of the resource group which the app service belongs to.

```

Comment:
Could we use resource phrasing here? 
`The name of the resource group in which the Automation Account is created.`

Reply:
should this also be force new?